### ISSUE TITLE
[BUG] Figure p in column(gridplot([[p]])) not rendered when adding another figure like this: column(p, gridplot([[p]]))

### ISSUE THREAD

**Issue by salsasepp** [bokeh/bokeh#12479]
### Software versions

Python version      :  3.10.6 (main, Aug 10 2022, 11:40:04) [GCC 11.3.0]
IPython version     :  (not installed)
Tornado version     :  6.2
Bokeh version       :  3.0.0.dev20
BokehJS static path :  /home/stefan/.local/lib/python3.10/site-packages/bokeh/server/static
node.js version     :  v16.16.0
npm version         :  8.19.2
Operating system    :  Linux-5.15.0-48-generic-x86_64-with-glibc2.35

### Browser name and version

Google Chrome Version 106.0.5249.103 (Official Build) (64-bit) (Ubuntu 22.04)

### Expected behavior

Line plots rendered normally similar to this screenshot
`show(column(gridplot([[plot1, plot2]])))`:
![Screenshot from 2022-10-10 15-07-28](https://user-images.githubusercontent.com/4884807/194873477-f1062216-3127-4dc1-bcd0-0b8d7a285242.png)

### Observed behavior

Figures within a `gridplot()` get messed up when embedded into a `column()` along with another plot, as in this screenshot. No glyph and no axes are visible. Note combining a div or other elements like a slider with the `gridplot()` works as expected, but using any `figure()` element generates the issue.

`show(column(gridplot([[plot1, plot2]]), plot1))`:
![Screenshot from 2022-10-10 15-09-03](https://user-images.githubusercontent.com/4884807/194873745-658deabb-d77b-4cc1-abf5-1ecce4f440c2.png)

Also note that the `basic/layouts/anscombe.py` example that uses a similar layout of a gridplot within a column can be edited to show the same issue. Replace the last line in anscombe.py
`show(column(div, grid, sizing_mode="scale_width"))`
with
`show(column(I, grid, sizing_mode="scale_width"))
`
### Example code

```Python
from bokeh.models import ColumnDataSource, Div
from bokeh.plotting import figure, show
from bokeh.layouts import column, gridplot

source = ColumnDataSource(data=dict(x=[0, 1], y=[0, 1]))
plot1 = figure()
plot1.line(source=source)
plot2 = figure()
plot2.line(source=source)

div = Div(text="This is a div")

# These work as expected
# show(column(gridplot([[plot1, plot2]])))
# show(column(div, gridplot([[plot1, plot2]])))
# show(column(plot1, plot2))

# These demonstrate the issue
show(column(plot1, gridplot([[plot1, plot2]])))
# show(column(gridplot([[plot1, plot2]]), plot1))
# show(column(gridplot([[plot1]]), plot1))
```

**Comment by mattpap** [comment]
This is another variant of lack of propagation of layout, that is specific to grid plots and similar layouts (tabs possibly). This should be easy to fix.

**Comment by salsasepp** [comment]
@mattpapp Great fix, thank you! In case #12465 is essentially the same thing and has a similarly easy fix, very much appreciated. #12465 currently breaks two apps of mine with bokeh 3.0.0rc4.
